graphql_compiler/tests/test_schema_generation.py

```diff
@@ -181,10 +221,46 @@ def test_link_property(self):
         self.assertEqual(out_property.qualifier, 'Person')
 
     def test_parsed_class_fields(self):
-        schema_data = [BASE_VERTEX_SCHEMA_DATA, ENTITY_SCHEMA_DATA,
-                       BASE_EDGE_SCHEMA_DATA, PERSON_SCHEMA_DATA,
-                       LOCATION_SCHEMA_DATA, PERSON_LIVES_IN_EDGE_SCHEMA_DATA]
+        schema_data = [
+            BASE_EDGE_SCHEMA_DATA,
+            BASE_VERTEX_SCHEMA_DATA,
+            ENTITY_SCHEMA_DATA,
+            LOCATION_SCHEMA_DATA,
+            PERSON_LIVES_IN_EDGE_SCHEMA_DATA,
+            PERSON_SCHEMA_DATA,
+        ]
         schema_graph = SchemaGraph(schema_data)
         person_lives_in_edge = schema_graph.get_element_by_class_name('Person_LivesIn')
         self.assertEqual(PERSON_LIVES_IN_EDGE_SCHEMA_DATA['customFields'],
                          person_lives_in_edge.class_fields)
+
+    def test_type_equivalence_dicts(self):
+        schema_data = [
+            BASE_EDGE_SCHEMA_DATA,
+            BASE_VERTEX_SCHEMA_DATA,
+            BABY_SCHEMA_DATA,
+            ENTITY_SCHEMA_DATA,
+            LOCATION_SCHEMA_DATA,
+            PERSON_LIVES_IN_EDGE_SCHEMA_DATA,
+            PERSON_SCHEMA_DATA,
+        ]
+        schema, type_equivalence_dicts = get_graphql_schema_from_orientdb_schema_data(schema_data)
+        person, person_baby_union = next(six.iteritems(type_equivalence_dicts))
+        baby = schema.get_type('Baby')
+        location = schema.get_type('Location')
+
+        # Assert that the Person class is part of the schema.
+        self.assertEqual(person, schema.get_type('Person'))
+
+        # Assert that the union consists of the Baby and Person classes
+        self.assertEqual(person_baby_union.types, [baby, person])
```

Comment:
Are we sure that the order of types in the union is always going to be this, and not the other way around? Would it be better to check this with set equality instead?